Week4/ReadingList/app/src/main/java/com/florencenjeri/readinglist/ui/BooksViewModelFactory.kt

```diff
@@ -0,0 +1,14 @@
+package com.florencenjeri.readinglist.ui
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import com.florencenjeri.readinglist.database.BooksRepository
+
+class BooksViewModelFactory(private val repository: BooksRepository) : ViewModelProvider.Factory {
+    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
+        if (modelClass.isAssignableFrom(BooksViewModel::class.java)) {
+            return BooksViewModel(repository) as T
```

Comment:
Okay so you used this, but you still had manual creation of the Repo in each Fragment/activity?

You could've created a repo in the App class, and then pass it in here instead. Or even create this factory in App, and use it to build viewModels, using the lazy delegate